cmd/thanos/sidecar.go

```diff
@@ -37,6 +37,8 @@ func registerSidecar(m map[string]setupFunc, app *kingpin.Application, name stri
 	promURL := cmd.Flag("prometheus.url", "URL at which to reach Prometheus's API. For better performance use local network.").
 		Default("http://localhost:9090").URL()
 
+	promRetention := modelDuration(cmd.Flag("prometheus.retention", "A limit on how much retention to query from Prometheus. 0d means query all TSDB data found on disk").Default("0d"))
```

Comment:
Curious about this. Should we do retention? I would rather see `storeapi.min-time` with similar model to specify both relative and absolute time as here: https://github.com/improbable-eng/thanos/pull/1077 

This is to keep it consistent with potential store gateway time partitioning.

Reply:
This was what I struggled with most -- how to name and handle this argument.  I'll take a look at implementing the suggestions.  I'm all about consistency.

Reply:
yy, if we are doing this then `--storeapi.min-time`, `--storeapi.max-time` sounds like good options.

You can copy paste the https://github.com/thanos-io/thanos/pull/1077/files#diff-dd29e6298d43e46bb651035051819cfcR14 class from my PR, to make it take same exact format.